annonatate/flaskserver.py

```diff
@@ -704,16 +704,6 @@ def parsetype(board):
     m = re.search(regex, board)
     return m.group(1)
 
-def updateAnnosGitHub():
-    annotations = session['annotations']
-    try:
-        githubresponse = github.get(session['currentworkspace']['contents_url'].replace('{+path}', filepath))
-        githubfilenames = list(map(lambda x: x['name'], githubresponse))
-        session['annotations'] = list(filter(lambda x: x['filename'].split('/')[-1] in githubfilenames,annotations))
```

Comment:
@dnoneill Another clarification request: why does this line use ```x['filename'].split('/')[-1]``` instead of just ```x['name']```? I'm finding that ```githubresponse``` is in the same annotation format as what's stored in ```session['annotations']```, so there are no filename attriibutes, but ```x['name']``` matches the name in ```githubfilenames```. Are there circumstances where this wouldn't be true? I've got a test now that uses ```x['name'] and shows that the local deletion of an annotation that is now missing on Github works.

Reply:
Because we are comparing the annotations in the session, which don't have a key that is name, they just have the full url which is under the key filename. While the GitHub files we found, which have the key name are in the format where they are just the filename ('xxxx-list.json', 'xxxxx.json') so we need to the the portion that we can match on. I am not quite sure what that test is showing but session annotations never have the key "name". You can see that in the API, https://dnoneill.github.io/annonatate/, which has the fields: filename, json and order.

Reply:
OK, thanks - I need to fix my test data. I'm giving it github annotations instead of session annotations.

Reply:
Ah. That makes sense. Glad that was semi coherent. I was worried you might not understand what I was talking about because it seemed rambly to me.